layouts/partials/nav_toggle.html

```diff
@@ -21,10 +21,33 @@
     </button>
   {{ end }}
   <div class="wrapper-logo-mobile">
-    {{- with .Page.Params.logo | default .Site.Params.logo | default "https://www.eclipse.org/eclipse.org-common/themes/solstice/public/images/logo/eclipse-foundation-white-orange.svg" }}
-      <a class="navbar-brand visible-xs" title="{{ $.Site.Title }}" href="{{ "" | absLangURL }}">
-        <img width="{{ $.Page.Params.logo_width | default $.Site.Params.logo_width | default "140"}}" class="logo-eclipse-default-mobile img-responsive" src="{{ . | absURL }}" alt="{{ $.Site.Title }}" />
+
+    {{ $flag := false }}
+    {{ $currentPageLogo := '0' }}
+    {{ $currentPageLogoTitle := '0' }}
+    {{ $currentPageLogoLink := '0' }}
+    {{ range $.Site.Data.subsites_custom_logo.items }}
+
+      {{ $section := $.Site.GetPage (.page_url) }}
+      {{ if or (eq ($.Section) ($section)) ( $.Page.IsDescendant $section ) }}
+        {{ $flag = true }}
+        {{ $currentPageLogo = .logo }}
+        {{ $currentPageLogoTitle = .logo_title }}
+        {{ $currentPageLogoLink = .logo_url }}
+      {{ end }}
+    {{ end }}
+
+    {{ if and (eq $flag true) (ne $currentPageLogo '0') }}
```

Comment:
why duplicate this section and use 0 values? We could set the default values of these fields to the current defaults, and override them if we find matches. Then we don't need a flag, and just set via the variables with less duplication

Reply:
Absolutely correct! I have no idea what was in my head when I wrote this. 😅

Updated with much cleaner code!